### DISCUSSION TITLE
application/x-www-form-urlencoded

### DISCUSSION THREAD

**Discussion by ldecourcel** [httprb/http#711] [Q&A] [answered]
Hello,
I am using http gem and for the first I encountered the application/x-www-form-urlencoded content-type.

I am supposed to make a post request sending a file but I cannot any ways to properly encode it and send it.

If anyone has an idea on how to proceed, it would be greatly appreciated if you share it.

Thanks !

**Comment by tarcieri** [comment, marked as answer]
Please see https://github.com/httprb/form_data

**Comment by ixti** [comment]
``` ruby
HTTP.post("http://example.com/resource", :form => {
  :username => "ixti",
  :avatar   => HTTP::FormData::File.new("/home/ixti/avatar.png")
})
```

See Also: https://github.com/httprb/http/wiki/Passing-Parameters
